consensus/hotstuff/verification/common.go

```diff
@@ -68,24 +69,42 @@ func checkVotesValidity(votes []*model.Vote) error {
 	return nil
 }
 
+type aggregate struct {
+	lastStakingSigners map[flow.Identifier]*flow.Identity
+	lastStakingKey     crypto.PublicKey
+}
+
 // stakingKeysAggregator is a structure that aggregates the staking
 // public keys for QC verifications.
 type stakingKeysAggregator struct {
-	lastStakingSigners map[flow.Identifier]*flow.Identity
-	lastStakingKey     crypto.PublicKey
-	sync.RWMutex
+	current unsafe.Pointer // *aggregate type
+	inner   *aggregate
```

Comment:
I think having `inner` here later can cause some complications. ( confusing )

Reply:
Maybe a name change or comment would help?  What would you like to see ?